api/src/utils/media/validation/validation-schema-parser.ts

```diff
@@ -0,0 +1,226 @@
+    const validationSchemas = this.getFileValidationSchemas(fileName);
+
+    const rules: CSVValidator[] = [];
+
+    validationSchemas.forEach((validationSchema) => {
+      const keys = Object.keys(validationSchema);
+
+      if (keys.length !== 1) {
+        return;
+      }
+
+      const key = keys[0];
+
+      const generatorFunction = ValidationRulesRegistry.findMatchingRule(key);
+
+      if (!generatorFunction) {
+        return;
+      }
+
+      const rule = generatorFunction(validationSchema);
+
+      rules.push(rule);
+    });
+
+    return rules;
+  }
+
+  getAllColumnValidations(fileName: string): CSVValidator[] {
+    const columnNames = this.getColumnNames(fileName);
+
+    let rules: CSVValidator[] = [];
+
+    columnNames.forEach((columnName) => {
+      const columnValidators = this.getColumnValidations(fileName, columnName);
+      rules = rules.concat(columnValidators);
+    });
+
+    return rules;
+  }
+
+  getColumnValidations(fileName: string, columnName: string): CSVValidator[] {
+    const validationSchemas = this.getColumnValidationSchemas(fileName, columnName);
+
+    const rules: CSVValidator[] = [];
+
+    validationSchemas.forEach((validationSchema) => {
+      const keys = Object.keys(validationSchema);
+
+      if (keys.length !== 1) {
+        return;
+      }
+
+      const key = keys[0];
+
+      const generatorFunction = ValidationRulesRegistry.findMatchingRule(key);
+
+      if (!generatorFunction) {
+        return;
+      }
+
+      const rule = generatorFunction({ columnName: columnName, ...validationSchema });
+
+      rules.push(rule);
+    });
+
+    return rules;
+  }
+
+  getSubmissionValidationSChemas(): object[] {
+    return jsonpath.query(this.validationSchema, this.getSubmissionValidationsJsonPath())?.[0] || [];
+  }
+
+  getFileValidationSchemas(fileName: string): object[] {
+    return jsonpath.query(this.validationSchema, this.getFileValidationsJsonPath(fileName))?.[0] || [];
+  }
+
+  getColumnValidationSchemas(fileName: string, columnName: string): { name: string; validations: [] }[] {
+    return jsonpath.query(this.validationSchema, this.getColumnValidationsJsonpath(fileName, columnName))?.[0] || [];
+  }
+
+  getColumnNames(fileName: string): string[] {
+    return jsonpath.query(this.validationSchema, this.getColumnNamesJsonpath(fileName));
+  }
+
+  getSubmissionValidationsJsonPath(): string {
+    return '$.validations';
+  }
+
+  getFileValidationsJsonPath(fileName: string): string {
+    return `$.files[?(@.name == '${fileName}')].validations`;
+  }
+
+  getColumnNamesJsonpath(fileName: string): string {
+    return `$.files[?(@.name == '${fileName}')].columns[*].name`;
+  }
+
+  getColumnValidationsJsonpath(fileName: string, columnName: string): string {
+    return `$.files[?(@.name == '${fileName}')].columns[?(@.name == '${columnName}')].validations`;
+  }
+
+  parseJson(json: any): object {
+    let parsedJson;
+
+    try {
+      parsedJson = JSON.parse(json);
```

Comment:
question: can we just do `return JSON.parse(json)` in the try and then if it doesnt work, then it will catch?
or does it have to be done this way where the return is outside the try/catch block?

Reply:
Yeah we can, good catch.  If there wasn't a throw in the catch, then it would complain about no return at the end.  But it appears smart enough to know it will always return an object OR throw.